### ISSUE TITLE
qgs/qgz project does not save mesh symbology correctly

### ISSUE THREAD

**Issue by desertstsung** [qgis/QGIS#54384]
### What is the bug or the crash?

project does not save mesh symbology settings correctly
Attach zipped file (10 m wind from ECMWF):[10uv.zip](https://github.com/qgis/QGIS/files/12451369/10uv.zip)

### Steps to reproduce the issue

1. drag the grb2 file into QGIS window
2. select the mesh item only, and click `Add Layers`
3. the canvas shows contours by default
4. right click layer name, go to `Properties`->`Symbology`->`Datasets` tab
5. click on the arrow symbol, which seems to be bolden, and click off the rectangle rainbow symbol, which seems to be grayscale
6. click apply
7. the canvas shows wind field
8. save the project in qgs or qgz format, and exit QGIS
9. open the saved project, nothing is on the canvas after loading, `Properties`->`Symbology`->`Datasets` tab shows both contour or vector is turned off, which the vector should be on accoording to step 5.

### Versions

3.32.1-1 under Arch Linux:
QGIS version | 3.32.1-Lima | QGIS code branch | Release 3.32
-- | -- | -- | --
Qt version | 5.15.10
Python version | 3.11.3
GDAL/OGR version | 3.7.0
PROJ version | 9.2.1
EPSG Registry database version | v10.088 (2023-05-13)
GEOS version | 3.12.0-CAPI-1.18.0
Compiled against SQLite | 3.42.0 | Running against SQLite | 3.43.0
Compiled against PDAL | 2.5.5 | Running against PDAL | 2.5.6
PostgreSQL client version | 15.3
SpatiaLite version | 5.0.1
QWT version | 6.2.0
QScintilla2 version | 2.14.1
OS version | Arch Linux
  |   |   |  
Active Python plugins
OSMDownloader | 1.0.3
MetaSearch | 0.3.6
db_manager | 0.1.20
grassprovider | 2.12.99
processing | 2.12.99

and 3.32.2 under Windows:
```
QGIS version
3.32.2-Lima
QGIS code revision
c0b8833964
Qt version
5.15.3
Python version
3.9.5
GDAL/OGR version
3.7.1
PROJ version
9.2.1
EPSG Registry database version
v10.088 (2023-05-13)
GEOS version
3.12.0-CAPI-1.18.0
SQLite version
3.41.1
PDAL version
2.5.5
PostgreSQL client version
15.2
SpatiaLite version
5.1.0
QWT version
6.1.6
QScintilla2 version
2.13.1
OS version
Windows 10 Version 2009

Active Python plugins
db_manager
0.1.20
grassprovider
2.12.99
MetaSearch
0.3.6
processing
2.12.99
```

### Supported QGIS version

- [x] I'm running a supported QGIS version according to [the roadmap](https://www.qgis.org/en/site/getinvolved/development/roadmap.html#release-schedule).

### New profile

- [X] I tried with a new [QGIS profile](https://docs.qgis.org/latest/en/docs/user_manual/introduction/qgis_configuration.html#working-with-user-profiles)

### Additional context

I'm here to seek for an alternative python solution, rather than a later bug-fix, or this is not a bug at all.
I have checked out that I can get the mesh layer by QgsProject.instance().mapLayer method, but I'm too noob to find out how to modify the vector/contour setting (step 5 in the above bug report) from the python console.
Hope you can give me more information on this question.
Thank you.

**Comment by desertstsung** [comment]
OK, I just found a temp solution: before save the project, set both vector/contour on, and adjust the opacity of contour to 0. Then when we open the saved project, the canvas only remains the arrow.

It could be a little more specific that I found this bug may due to the `style` model, since when I change the style name or add a style in the `Symbology` subwindow, and save the project, I cannot grep any new style name in the qgs xml text.

Hope the info I provided could help you to improve QGIS :)
